packages/forms/src/patterns/select-dropdown.ts

```diff
@@ -45,9 +58,34 @@ export const selectDropdownConfig: PatternConfig<
       { value: 'value3', label: 'Option-3' },
     ],
   },
-  //  STILL IN PROGRESS:
-  parseUserInput: (_, inputObj) => {
-    return safeZodParseToFormError(SelectDropdownSchema, inputObj);
+
+  parseUserInput: (pattern, inputObj) => {
+    const expectedInput = inputObj as { value: string };
+
+    const schema = createSchema(pattern.data);
+    try {
+      const parsedValue = schema.parse(expectedInput.value);
+      return parsedValue
+        ? { success: true, data: parsedValue }
+        : {
+            success: false,
+            error: {
+              type: 'custom',
+              message: 'Parsed select dropdown value is undefined',
+            },
+          };
+    } catch (e) {
+      const zodError = e as z.ZodError;
+      return {
+        success: false,
+        error: {
+          type: 'custom',
+          message: zodError.errors
+            ? zodError.errors[0].message
+            : zodError.message,
+        },
+      };
+    }
```

Comment:
Would the `safeZodParseToFormError()` function work here, to simplify this logic?

Reply:
I tried but, it seems safeZodParseToFormError() doesn't work well with enum.

Reply:
It turns out issue was not related safeZodParseToFormError() not being able to handle the enum, but I was passing incorrect data type (object) from my tests. When input type (string) passed it works fine. Refactored the hook and updated tests accordingly. Thank you for syncing with me earlier today!